frontend/gateway/grpcclient/client.go

```diff
@@ -28,6 +28,8 @@ type GrpcClient interface {
 }
 
 func New(ctx context.Context, opts map[string]string, session, product string, c pb.LLBBridgeClient, w []client.WorkerInfo) (GrpcClient, error) {
+	ctx, cancel := context.WithTimeout(ctx, 5*time.Second)
```

Comment:
should this be set by the caller?

Reply:
This is only for the Ping request to check that connection is active. This part isn't actually ever reached with the new error handling. I added it for the case if someone is manually running frontend with an unresponsive configuration.